flink-tests/src/test/java/org/apache/flink/test/checkpointing/SavepointITCase.java

```diff
@@ -339,9 +339,12 @@ protected void run() {
 					SubtaskState subtaskState = taskState.getState(tdd.getIndexInSubtaskGroup());
 
 					assertNotNull(subtaskState);
-					errMsg = "Initial operator state mismatch.";
-					assertEquals(errMsg, subtaskState.getLegacyOperatorState(),
-							tdd.getTaskStateHandles().getLegacyOperatorState());
+
+					if (subtaskState.hasState()) {
```

Comment:
I think this check can actually be replaced by `subtaskState.getLegacyOperatorState()` and if you do that there is no place in the code that calls `SubtaskState.hasState()`, making it obsolete.

Please correct me if I'm wrong. 😅